staging/src/k8s.io/cloud-provider/sample/basic_main.go

```diff
@@ -41,6 +41,15 @@ import (
 )
 
 func main() {
+	err := runCloudControllerManagerCmd()
+	defer func() {
```

Comment:
The defer need to be registered prior to the call to runCloudControllerManagerCmd to be sure it will be called if an exception is thrown.

Reply:
correct me if not. since the logic is quite simple, i called the defer func after the `runCloudControllerManagerCmd`, if exception like `panic` happens, defer func will not be called also as I didn't recover the panic. Or you think it should still be refactored as 
```
func main() {
	var err error
	defer func() {
		if err != nil {
			os.Exit(1)
		}
	}()
	err = runCloudControllerManagerCmd()
}
```

Reply:
+1 on @kerthcet 
And IMO we don't even have to use defer func for it. 
https://github.com/kubernetes/kubernetes/pull/104753#discussion_r705121978

Reply:
I was more thinking something like
```
func main() { 
    var err error 
    defer func() {
        if stack := recover(); stack != nil {
            klog.Warningf("Exception from runCloudControllerManagerCmd %v", stack)
            os.Exit(2)
        } 
         if err != nil { 
            os.Exit(1) 
        } 
    }() 
    err = runCloudControllerManagerCmd() 
}
```

Reply:
IMO, `panic` will also collect the stack traces,  and if panic happens, we will always exit the thread, I thought there's no need to recover these panics, or maybe I missed something and you can give me some advices. 🧐 what's more,  I agree with the `defer` func to ensure we will handle this errors finally.